crates/iota-types/src/digests.rs

```diff
@@ -183,6 +186,25 @@ static IOTA_PROTOCOL_CONFIG_CHAIN_OVERRIDE: Lazy<Option<Chain>> = Lazy::new(|| {
 });
 
 impl ChainIdentifier {
+    /// Take a short 4 byte identifier and convert it into a ChainIdentifier.
+    /// Short ids come from the JSON RPC getChainIdentifier and are encoded in
+    /// hex.
+    pub fn from_chain_short_id(short_id: &String) -> Option<Self> {
```

Comment:
think it's helpful to make it a bit more flexible with a `impl AsRef<str> + std::fmt::Display` for example>?

Also in Line 195 and 200 you could just inline the `short_id`.

Reply:
#5488